common/src/main/java/org/dash/wallet/common/util/NavigationExtensions.kt

```diff
@@ -10,18 +10,24 @@ import androidx.navigation.fragment.findNavController
 // Avoids "Navigation action/destination xyz cannot be found from the current destination" errors
 fun Fragment.safeNavigate(directions: NavDirections) {
     val navController = findNavController()
-    val destination = navController.currentDestination as FragmentNavigator.Destination
 
-    if (javaClass.name == destination.className) {
-        navController.navigate(directions)
+    if (navController.currentDestination is FragmentNavigator.Destination) {
+        val destination = navController.currentDestination as FragmentNavigator.Destination
+
+        if (javaClass.name == destination.className) {
+            navController.navigate(directions)
+        }
     }
 }
 
 fun DialogFragment.dialogSafeNavigate(directions: NavDirections) {
     val navController = findNavController()
-    val destination = navController.currentDestination as DialogFragmentNavigator.Destination
 
-    if (javaClass.name == destination.className) {
-        navController.navigate(directions)
+    if (navController.currentDestination is DialogFragmentNavigator.Destination) {
```

Comment:
Here we need to check that `currentDestination` is of the correct type, otherwise might cause a crash for the same reason.